### DISCUSSION TITLE
CD-ROM driver for 8088? (Closed)

### DISCUSSION THREAD

**Discussion by jadasse** [86Box/86Box#2695] [General]
Are there any CD-ROM drivers for 8088? ATAPI or SCSI? I have a Xi8088 mobo with a megabyte of RAM and with XTIDE to handle my hard drive, and I am considering adding disc support, so should I use ATAPI or SCSI, and what drivers are compatible? The Microsoft MSCDEX installer that installs 2.23 only works for at least a 386.

**Comment by TC1995** [comment]
2.23 works on 8088 too, though it's best to use an SCSI controller on 8088/8086's, namely the Trantor T128 (ma120.sys), longshine/corel ls2000/rancho (aspils_d.sys). Trantor T130B can also run on a 8088, but without hdd support since it relies on v20/186 instructions.

**Reply by jadasse** [reply]
Do you have any tips where I can find compatible floppy disk drivers?

**Comment by jadasse** [comment]
The installer for for MSCDEX isn't compatible though:
[MSCDEX 2.23 Installer (incl. VIDE-CDD).ima.zip](https://github.com/86Box/86Box/files/9639359/MSCDEX.2.23.Installer.incl.VIDE-CDD.ima.zip)

**Comment by jadasse** [comment]
Is there a floppy disk driver I can use that is compatible with 8088?

**Reply by dhrdlicka** [reply]
Why would you need a floppy disk driver?

**Reply by jadasse** [reply]
*sigh* I meant a driver for the CD-ROM that is a floppy disk (.img, .ima)

**Reply by jadasse** [reply]
https://winworldpc.com/product/ms-cd-extensions-msc this doesn't work on 8088, I need a 386 for this

**Comment by jadasse** [comment]
http://www.minuszerodegrees.net/backpack_cdrom/backpack_cdrom.htm These drivers don't work

**Reply by jriwanek** [reply]
Not a surprise, the backpack connects via parallel port, not IDE or SCSI, we on't emulate a parallel port CD drive currently.

**Comment by TC1995** [comment]
You can manually edit config.sys/autoexec.bat.

**Reply by jadasse** [reply]
what drivers do you recommend?

**Reply by jadasse** [reply]
for instance, which specific .sys files are compatible with xt?

**Reply by guest123guest** [reply]
> for instance, which specific .sys files are compatible with xt?

I know the first CD-ROM drive was a Phillips drive that might have been designed for an XT class machine, maybe 86box supports emulating that and using the drivers for it?

**Comment by guest123guest** [comment]
Why would you want CD-ROM support for 8088 machines? Wouldn't that just be a novelty?

**Reply by jriwanek** [reply]
Not really, no, people did infact do that sort of thing.

**Reply by guest123guest** [reply]
> Not really, no, people did infact do that sort of thing.

Aside from a few Microsoft titles released on CD-ROM that are basically just older versions of encyclopedias they made, would there really be any killer apps for CD-ROM drives on an XT machine?

**Reply by jadasse** [reply]
> > Not really, no, people did infact do that sort of thing.
> 
> 
> 
> Aside from a few Microsoft titles released on CD-ROM that are basically just older versions of encyclopedias they made, would there really be any killer apps for CD-ROM drives on an XT machine?

I want play cue files on the Sound Blaster Pro CD player. Please don't judge me further.

**Reply by guest123guest** [reply]
> > > Not really, no, people did infact do that sort of thing.
> > 
> > 
> > Aside from a few Microsoft titles released on CD-ROM that are basically just older versions of encyclopedias they made, would there really be any killer apps for CD-ROM drives on an XT machine?
> 
> I want play cue files on the Sound Blaster Pro CD player. Please don't judge me further.

Why wouldn't that work on something newer like a 486? I might suggest just playing the cue file in an audio player on the host but I think bin/cue support in something like VLC might be weird enough to not work.

**Reply by jadasse** [reply]
> > > > Not really, no, people did infact do that sort of thing.
> 
> > > 
> 
> > > 
> 
> > > Aside from a few Microsoft titles released on CD-ROM that are basically just older versions of encyclopedias they made, would there really be any killer apps for CD-ROM drives on an XT machine?
> 
> > 
> 
> > I want play cue files on the Sound Blaster Pro CD player. Please don't judge me further.
> 
> 
> 
> Why wouldn't that work on something newer like a 486? I might suggest just playing the cue file in an audio player on the host but I think bin/cue support in something like VLC might be weird enough to not work.

It does! I just want to experiment Audio CD playback on an 8088 processor to see if it's feasible. It must be!

**Reply by jadasse** [reply]
I just need compatible .sys files or an installer I can use for the drive to work.

**Comment by jadasse** [comment]
Can someone PLEASE give me tips in how I can find a compatible driver for IBM XT? Or can they reply a link to a compatible driver?

**Comment by Torinde** [comment]
FreeDOS has a kind of 8086 edition - not sure if their MSCDEX equivalent runs there.

**Reply by jadasse** [reply]
I'm also looking for an accompanying .sys driver that is also compatible with 8086 type CPUs. Unfortunately I've had no luck as I'm getting no response upon trying to put an image file in.

**Comment by TC1995** [comment]
CorelSCSI 2.0 has drivers for it, just look for aspils_d.sys (any type of cpu, including 8086 and 8088) and its cuni_asp.sys.
And yes, MSCDEX works on 808x too.
aspils_d.sys is mainly for the Corel LS2000 SCSI controller and other MMIO 53c400-based cards (Including Rancho and Longshine).

**Reply by jadasse** [reply]
ASPILS_D.SYS says no Corel SCSI cards were detected, although I have a Corel LS2000, IRQ 3, and address D800H.

**Reply by TC1995** [reply]
you need to use /LS2000,d800,3 then.

**Reply by jadasse** [reply]
Everything works! AT LAST! Thank you so much!

**Reply by TC1995** [reply]
> Everything works! AT LAST! Thank you so much!

You're welcome.

**Reply by jadasse** [reply]
It works again :P

**Reply by OBattler** [reply]
Just, by the way, I now have my XTIDE CD-ROM driver I made based on the open-source PATACD.SYS. I should upload it somewhere, perhaps even on GitHub.
